test/images/network-tester/webserver.go

```diff
@@ -227,32 +228,46 @@ func contactOthers(state *State) {
 		log.Printf("Server version: %#v\n", v)
 	}
 
+
+	for start := time.Now(); time.Since(start) < waitTimeout; time.Sleep(10 * time.Second) {
+		eps := getWebserverEndpoints()
+		if len(eps) < *peerCount {
+			state.Logf("%v/%v has %v endpoints, which is less than %v as expected. Waiting for all endpoints to come up.", *namespace, *service, len(eps), *peerCount)
+		}
+	}
+
 	// Do this repeatedly, in case there's some propagation delay with getting
 	// newly started pods into the endpoints list.
 	for i := 0; i < 15; i++ {
-		endpoints, err := client.Endpoints(*namespace).Get(*service)
-		if err != nil {
-			state.Logf("Unable to read the endpoints for %v/%v: %v; will try again.", *namespace, *service, err)
-			time.Sleep(time.Duration(1+rand.Intn(10)) * time.Second)
-		}
-
-		eps := sets.String{}
-		for _, ss := range endpoints.Subsets {
-			for _, a := range ss.Addresses {
-				for _, p := range ss.Ports {
-					eps.Insert(fmt.Sprintf("http://%s:%d", a.IP, p.Port))
-				}
-			}
-		}
+		eps := getWebserverEndpoints()
 		for ep := range eps {
 			state.Logf("Attempting to contact %s", ep)
 			contactSingle(ep, state)
 		}
-
 		time.Sleep(5 * time.Second)
 	}
 }
 
+//getWebserverEndpoints returns the webserver endpoints as a set of String, each in the format like "http://{ip}:{port}"
+func getWebserverEndpoints() sets.String {
+	client, err := client.NewInCluster()
```

Comment:
in general (unless you're sure) you shouldn't create http cliets in a loop. It will lead to a memory leak because clients use Conection:keepaive and are cached by the runtime. I mean plain old http clients, which underpins NewInCluster.